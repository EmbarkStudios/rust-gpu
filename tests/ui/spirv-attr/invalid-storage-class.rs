// Tests that certain storage class `#[spirv(...)]` attributes are disallowed.

// build-fail

#[spirv(vertex)]
fn _entry(
    #[spirv(input)] _: (),
    #[spirv(output)] _: (),
    #[spirv(private)] _: (),
    #[spirv(function)] _: (),
    #[spirv(generic)] _: (),
) {
}
