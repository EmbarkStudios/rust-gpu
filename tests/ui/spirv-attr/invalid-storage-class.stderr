error: `#[panic_handler]` function required, but not found

error: `Input` storage class is the default and should not be explicitly specified
 --> $DIR/invalid-storage-class.rs:7:13
  |
7 |     #[spirv(input)] _: (),
  |             ^^^^^

error: `Output` storage class is the default and should not be explicitly specified
 --> $DIR/invalid-storage-class.rs:8:13
  |
8 |     #[spirv(output)] _: (),
  |             ^^^^^^

error: `Private` storage class can not be used as part of an entry's interface
 --> $DIR/invalid-storage-class.rs:9:13
  |
9 |     #[spirv(private)] _: (),
  |             ^^^^^^^

error: `Function` storage class can not be used as part of an entry's interface
  --> $DIR/invalid-storage-class.rs:10:13
   |
10 |     #[spirv(function)] _: (),
   |             ^^^^^^^^

error: `Generic` storage class can not be used as part of an entry's interface
  --> $DIR/invalid-storage-class.rs:11:13
   |
11 |     #[spirv(generic)] _: (),
   |             ^^^^^^^

error: aborting due to 6 previous errors

